### PR TITLE
feat(buttongroup): add the pf buttongroup

### LINKED ISSUE #12: Component - Button and Button Group
Implement components that provides the following UI elements:

- ~~[PatternFly Button](http://www.patternfly.org/pattern-library/widgets/#buttons)~~ *This component as already been contributed*
- [PatternFly Button Group](http://www.patternfly.org/pattern-library/widgets/#button-groups)

Acceptance Criteria:

- When rendered, the component matches the examples in the PatternFly Pattern Library (see link above)
- the component supports:
  - color variations as shown in the PatternFly library
  - size variations as shown in the PatternFly library
  - the disabled state

### PR DESCRIPTION
**Pull request by dabeng**
**What**:
This PR adds the PatternFly buttongroup patterns. To be used to close #12 (ButtonGroup) once done.

Storybook: https://rawgit.com/dabeng/patternfly-react/btn-group-storybook/.out/index.html?knob-Label=Danger%20Will%20Robinson%21&selectedKind=Button&selectedStory=ButtonGroup&full=0&down=1&left=1&panelRight=0&downPanel=storybooks%2Fstorybook-addon-knobs

**How**:
Encapsulating react-bootstrap buttongroup as a story of patternfly-react storybook.

@priley86 , please review.

### CODE REVIEW

File: src/ButtonGroup/ButtonGroup.stories.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    documentationLink:
+      'http://www.patternfly.org/pattern-library/widgets/#button-groups',
+    description: description
+  })
+)
+
+stories.addWithInfo('ButtonGroup', () => {
+  return (
+    <div>
+      <div className="row" style={{ marginBottom: '20px' }}>
```
Comment (priley86) [inline src/ButtonGroup/ButtonGroup.stories.js]: We will need to rebase this PR w/ #77 once `Row` is available.

**Comment by priley86** [comment]
@dabeng thanks for pulling this together. I think `ButtonGroup` stories/components can exist within `src/Button` as they are so closely related. We are quickly expanding the left hand storybook nav and have not adopted chapters yet. @jgiardino thoughts on keeping ButtonGroup w/ Button stories?

File: src/ButtonGroup/ButtonGroup.stories.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        <div className="col-md-3">
```
Comment (priley86) [inline src/ButtonGroup/ButtonGroup.stories.js]: Same - this becomes `Col`

**Comment by jgiardino** [comment]
@priley86 Yes, I like that idea! Grouping similar components like button group under button sounds good to me.

**Comment by jgiardino** [comment]
Aside from @priley86 comments, looks good to me!

Thanks for adding this!

**Comment by priley86** [comment]
needs a rebase here and a few modifications when you return @dabeng. Hope everything is ok there!

**Comment by dabeng** [comment]
Hi @priley86 , I have introduced component Grid into this PR and rebased it with latest master branch.

**Comment by priley86** [comment]
great job! thanks @dabeng.

File: src/components/Button/ButtonGroup.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+import { ButtonGroup } from 'react-bootstrap';
+export default ButtonGroup;
```
Comment (priley86) [inline src/components/Button/ButtonGroup.js]: actually - this can now be simply:
`export { ButtonGroup as default } from 'react-bootstrap';`

Comment (priley86) [reply src/components/Button/ButtonGroup.js]: would you mind sending me a small patch (using `style(buttongroup): change export style`) commit header?

Comment (priley86) [reply src/components/Button/ButtonGroup.js]: let me know if there is any issue @dabeng

**Pull request merged**